### DISCUSSION TITLE
PlugData v0.5

### DISCUSSION THREAD

**Discussion by timothyschoen** [plugdata-team/plugdata#68] [General]
New version with many new features and bug fixes.

This is not production ready yet, and considering the large number of changes, there will probably still be some bugs. Let me know if you find any!

IMPORTANT: 
If you've used previous versions of PlugData, it is recommended to refresh your externals folder. To do this, remove the "Abstractions" and "Documentation" folders from the following location:
MacOS: ~/Library/PlugData/
Linux: ~/.config/PlugData/
Windows: %APPDATA%\PlugData\

Because there are many changes in this version, it breaks compatibility with older versions. If you are using an older version of PlugData in a DAW project, you'll have to save the pd patch, remove PlugData from the project, upgrade PlugData and re-add PlugData in the project. I'll start providing legacy support once we're past the alpha stage.

Features:
- Relative grid for aligning objects
- Light mode for better pd compatibility
- New look
- A DAW transport object: [playhead]
- Settable tail length for plugins
- Undo/Redo for changing connection paths (work-in-progress)
- Updated ELSE to 1.0-rc1

Fixes:
- Many improvements to pd patch rendering
- Fixed DAW crashes when saving/loading presets
- Fixed fullscreen bug
- Fixed externals not loading on Linux
- PlugDataInfo comment now hidden in pd patches
- Clearer feedback on temporary lock
- Using to front now actually moves the object to front in pd patch (warning: this can impact the order of evaluation)
- Fixed wrong number of radiobox elements
- Fixed crash when renaming objects
- Fixed crash on tab close
- Many other fixes!

Thanks to @ludnny and @FlachyJoe for reporting many of these bugs!

Known Issues:
- Externals are broken on Windows
- Can only open 1 project at a time, it can open (but not save) help files and abstractions.
- Problems with Bitwig and Cubase (at least with earlier versions)

- Broken Pd objects:
  - [text define]
  
- Broken ELSE objects:
  - [canvas.edit] (and possibly more canvas related objects)
  - [store]
  - [rec]

<img width="1054" alt="Screenshot 2022-04-21 at 03 35 01" src="https://user-images.githubusercontent.com/44585538/164355204-7b81a7ea-d6e7-46d5-9866-cbc3a77a4b46.png">

<img width="1054" alt="Screenshot 2022-04-21 at 03 30 17" src="https://user-images.githubusercontent.com/44585538/164355197-d6b03403-21f2-498b-9110-7a6757be4c3a.png">

<hr /><em>This discussion was created from the release <a href='https://github.com/timothyschoen/PlugData/releases/tag/v0.5'>PlugData v0.5</a>.</em>

**Comment by ludnny** [comment]
Thank you very much!
Amazing release!

**Reply by timothyschoen** [reply]
Thanks :D

**Comment by jaffasplaffa** [comment]
Hey :) 

I just wanted to ask about the DAW transport object. It says it's called "playlist" in the release notes, but it does not show up when I search for it. However a "playhead" comes up? Is that it?

I tried right clicking "playhead" and select help, to see if there was a help file, so I could check if it was the right object, but it opens "canvas-help.pd" patch.

Then I tried another object and right clicked and selected help. It seems like for some object, if you right click and select help, you are taken to the same help file, the "canvas-help.pd". Maybe this happens when trying to acces a help file that does not exist?

Best wishes,

Jakob

**Reply by timothyschoen** [reply]
Oops that's a mistake, thanks for letting me know! It's called playhead indeed, I've fixed the typo.

**Reply by jaffasplaffa** [reply]
No problem :) 

I tested it a bit and made something like a help file, that shows what the different outputs are. Some of them was easy to figure out, but some of them I am not sure about. I can't upload pd patches here, but if you can use it for anything, i'd be happy to supply it :) 

<img width="998" alt="Screenshot 2022-04-24 at 16 59 11" src="https://user-images.githubusercontent.com/14827192/164982682-1da353af-6ceb-49ce-a926-9cd9876c0298.png">

**Reply by timothyschoen** [reply]
Very nice! If you could mail it to me: timschoen123@gmail.com I'll include it in the next version!

**Reply by jaffasplaffa** [reply]
I just send it to you :)

**Reply by ludnny** [reply]
@jaffasplaffa For the other outputs, you can see their names directly in the _playhead_ abstraction (Right-click / Open):

![image](https://user-images.githubusercontent.com/46849877/165087833-f853e83f-1d42-457d-935e-de2299f7ae2f.png)

**Reply by jaffasplaffa** [reply]
Thanks Ludnny.

I did not know it was an abstraction, i thought it was an external. But I see it now. My bad, I apologise.

**Comment by alfonso73** [comment]
Great @timothyschoen !
Could the following be interesting for 0.6? ;-)

https://github.com/pierreguillot/Camomile/issues/242#issuecomment-1103273871

**Reply by timothyschoen** [reply]
I am, at least for the plugin version. Unfortunately it's not "switchable" so it can't be optional, and also breaks compatibility with externals. Externals don't work in the plugin anyways, so might as well implement this! See #70 btw
